pkg/events/derive/derive.go

```diff
@@ -10,6 +11,9 @@ import (
 // Use as static variable for testability reasons
 var getEventDefFunc = events.Definitions.Get
 
+// InvokedContext is a map used to store the triggering events' context
+var InvokedContext = make(map[uint64]trace.Event)
```

Comment:
ditto on the mutex.
I'm also not sure if this is the best place to have the map, Maybe it should get it's own pacakge like most other caches, and then it can be managed there, with memory safety and all. It can be given as a dependency to the deriving functions.